### ISSUE TITLE
Rework AuthRepositoryImpl exception handling

### ISSUE THREAD

**Issue by thePlebDev** [thePlebDev/CalfTracker#348]
# Proposed 
- After reading [Programming Android with Kotlin](https://www.oreilly.com/library/view/programming-android-with/9781492062998/), I have realized that I have been doing exception handling improperly
- I need to adhere to the `Exception Transparency` principle 

# Why is this important 
- This will not only give my code separation of concerns. But, it will also stop my code from giving unwanted results when handling exceptions 

# Additional context
- Basically, I need to take all the try/catch blocks out of `flow{}` and move them to the [catch](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/catch.html) operator

**Comment by thePlebDev** [comment]
- I have refactored all of the `AuthRepositoryImpl` code to use the `catch` operator instead of the `try/catch` block
